osp/core/session/core_session.py

```diff
@@ -61,9 +61,11 @@ def close(self):
         def __iter__(self):
             """Iterate the result."""
             for row in self.result:
-                yield CoreSession.CoreSessionSparqlBindingSet(row,
-                                                              self.session,
-                                                              self.datatypes)
+                result = CoreSession.CoreSessionSparqlBindingSet(
+                    row, self.session, self.datatypes
+                )
+                self.datatypes = dict()
+                yield result
```

Comment:
@kysrpex: For resolving this thread, I do not see another possiblity than reassigning an empty `dict` right after iterating the `SparqlResult`. For sure, this is not very beautiful in terms of implementation since it needs to regarded each time while defining the `__iter__`-method. 

However, would you have another idea?

Reply:
@kysrpex, I just saw that this does not work if the `SparqlResult` returns more than one row (of course not - my fault). I will try to think of another solution.